app/test/integration_test.go

```diff
@@ -284,3 +284,73 @@ func queryTx(clientCtx client.Context, hashHexStr string, prove bool) (*rpctypes
 
 	return node.Tx(context.Background(), hash, prove)
 }
+
+func (s *IntegrationTestSuite) TestSharesInclusionProof() {
+	require := s.Require()
+	val := s.network.Validators[0]
+
+	// generate 100 randomly sized txs (max size == 100kb)
+	txs := blobfactory.RandBlobTxsWithAccounts(
+		s.cfg.TxConfig.TxEncoder(),
+		s.kr,
+		val.ClientCtx.GRPCClient,
+		100000,
+		true,
+		s.cfg.ChainID,
+		s.accounts[:20],
+	)
+
+	hashes := make([]string, len(txs))
+
+	for i, tx := range txs {
+		res, err := val.ClientCtx.BroadcastTxSync(tx)
+		require.NoError(err)
+		require.Equal(abci.CodeTypeOK, res.Code)
+		hashes[i] = res.TxHash
+	}
+
+	// wait a few blocks to clear the txs
+	for i := 0; i < 20; i++ {
+		require.NoError(s.network.WaitForNextBlock())
+	}
+
+	for _, hash := range hashes {
+		txResp, err := queryTx(val.ClientCtx, hash, true)
+		require.NoError(err)
+		require.Equal(abci.CodeTypeOK, txResp.TxResult.Code)
+
+		// verify that the transaction inclusion proof is valid
+		require.True(txResp.Proof.VerifyProof())
+
+		// get the transaction shares
+		node, err := val.ClientCtx.GetNode()
+		require.NoError(err)
+		blockRes, err := node.Block(context.Background(), &txResp.Height)
+		require.NoError(err)
+		beginTxShare, endTxShare, err := prove.TxSharePosition(blockRes.Block.Txs, uint64(txResp.Index))
+		require.NoError(err)
+
+		txProof, err := node.ProveShares(
+			context.Background(),
+			uint64(txResp.Height),
+			beginTxShare,
+			endTxShare,
+		)
+		require.NoError(err)
+		require.NoError(txProof.Validate(blockRes.Block.DataHash))
+
+		// get the message shares
+		beginMsgShare, endMsgShare, err := prove.MsgSharesPosition(blockRes.Block.Txs[txResp.Index])
```

Comment:
[nit] rename message to blob so something like:
```golang
		// get the blob shares
		beginBlobShare, endBlobShare, err := prove.BlobSharePositions(blockRes.Block.Txs[txResp.Index])
```

also consider renaming `MsgSharesPosition` => `BlobSharePositions` because the function appears to return two positions. Alternative naming proposal: `BlobShareRange`

Reply:
4595dd2